### ISSUE TITLE
Edit and retry functionality can fail when multiple messages are correlated

### ISSUE THREAD

**Issue by cquirosj** [Particular/ServicePulse#1570]
#### Description

While editing a message using the [edit and retry functionality](https://docs.particular.net/servicepulse/intro-editing-messages), if there are multiple correlated failed messages in the error queue, the editor dialog in ServicePulse could potentially load the incorrect message, making the editing procedure fail when the user submits the changes.

#### Expected behavior

When the user proceeds to edit the message: 
- An alert should get displayed in ServicePulse UI informing that is being retried: 'Retrying the message X'
- No warning errors being logged in ServiceControl related to the message just edited

#### Actual behavior

When the user proceeds to edit the message: 
- An error is displayed in ServicePulse UI with the following description: An error occurred while retrying the message, please check the ServiceControl logs for more details on the failure. 
- ServiceControl logs the following warning message: Warn|ServiceControl.MessageFailures.Api.EditFailedMessagesController|Locked headers have been modified on the edit-retry for MessageID

#### Versions
This is confirmed in version v1.32.4 and the latest at the moment of recording this bug( v1.37.0) This could be happening in all versions of ServicePulse, but more versions need to be checked to confirm this statement.

### Steps to reproduce

- Create an endpoint A that publishes an event.
- Create two different endpoints, Endpoint B and Endpoint C, and have them subscribe to the event published by Endpoint A. 
- Make the event handlers for each endpoint throw an exception.
- Configure the endpoints to use the platform.
- Using ServicePulse, inspect each failed message and locate the message with an incorrect value being displayed for the `NServiceBus.OriginatingEndpoint` header. For example, if the failed message was handled by endpoint A but `NServiceBus.OriginatingEndpoint` header shows endpoint B as the header's value.
- Use the edit and retry button to edit the body and click the retry button.

As an alternative, use the Saga Timeouts [sample](https://docs.particular.net/nservicebus/sagas/timeouts) and update OrderPlacedHandler.cs in the billing endpoint and ShippingPolicy.cs in the shipping endpoint to throw an exception where the OrderPlaced event is being handled

### Relevant log output

```shell
Warn|ServiceControl.MessageFailures.Api.EditFailedMessagesController|Locked headers have been modified on the edit-retry for MessageID
```

**Comment by nodranael** [comment]
Hello,

Is there progress on this issue ? This was created through customer support, and my company relies on this feature.

**Comment by lailabougria** [comment]
Hi there, @nodranael. We have this on our radar, and it is eligible to be picked up for a future release, but unfortunately, we cannot provide a firm timeline on this at the moment.

**Comment by nodranael** [comment]
Hello,

Any update on this ?

**Comment by danielmarbach** [comment]
Hi @nodranael 

I don't have additional information other than what @lailabougria already provided, I'm afraid. I'll try to get some more information and keep you posted.

Daniel

**Comment by jpalac** [comment]
Will be released in 1.40.1

**Comment by jpalac** [comment]
ServicePulse [1.41.0](https://github.com/Particular/ServicePulse/releases/tag/1.41.0) released
